internal/ethapi/api.go

```diff
@@ -73,27 +74,56 @@ func (s *PublicEthereumAPI) ProtocolVersion() hexutil.Uint {
 	return hexutil.Uint(s.b.ProtocolVersion())
 }
 
+//
+// FIXME: Can't get jsonschema oneOf to work right.
+//
+type EthSyncingResult struct {
```

Comment:
Perhaps you can infer this from the contents of the struct?

for example, when you encounter a struct, it  can be any of the following json schema types:
1. object
2. allOf
3. anyOf

As it turns out, any of these can be in effect, just an object. However, you figure out which to use based on:
- if all the keys in the struct are optional, it MAY be an 'anyOf'
- If all the keys in the struct are required, it MAY be an 'allOf'
- if there is some keys required, others not, it MUST be an object.

While following these heuristics only wont give a perfect solution, perhaps its worth a try?